src/components/ComparisonPortal/CountryComparisonAccordion.tsx

```diff
@@ -0,0 +1,73 @@
+'use client';
+
+import { useEffect, useMemo, useState } from 'react';
+
+import ComparisonAccordionSkeleton from '@/components/ComparisonPortal/ComparisonAccordionSkeleton';
+import { useSnackbar } from '@/domain/contexts/SnackbarContext';
+import { CountryMapData } from '@/domain/entities/country/CountryMapData';
+import { useCountryDataListQuery, useCountryIso3DataListQuery } from '@/domain/hooks/countryHooks';
+import { CountryComparisonOperations } from '@/operations/comparison-portal/CountryComparisonOperations';
+
+import AccordionContainer from '../Accordions/AccordionContainer';
+
+interface CountryComparisonAccordionProps {
+  selectedCountries: CountryMapData[] | undefined;
+}
+
+export default function CountryComparisonAccordion({ selectedCountries }: CountryComparisonAccordionProps) {
+  const [expandAll, setExpandAll] = useState(false);
+  const { showSnackBar } = useSnackbar();
+
+  const countryDataQuery = useCountryDataListQuery(
+    selectedCountries?.map((country) => country.properties.adm0_id),
+    (countryId) => {
+      if (!selectedCountries) return;
+      const countryName = CountryComparisonOperations.getCountryNameById(countryId, selectedCountries);
+      CountryComparisonOperations.showDataNotFoundSnackBar(showSnackBar, countryName);
+    }
+  );
+
+  const countryIso3DataQuery = useCountryIso3DataListQuery(
+    selectedCountries?.map((country) => country.properties.iso3),
+    (countryCode) => {
+      if (!selectedCountries) return;
+      const countryName = CountryComparisonOperations.getCountryNameByIso3(countryCode, selectedCountries);
+      CountryComparisonOperations.showDataNotFoundSnackBar(showSnackBar, countryName);
+    }
+  );
+
+  const isLoading = useMemo(() => {
+    return (
+      countryDataQuery.some((result) => result.isLoading) || countryIso3DataQuery.some((result) => result.isLoading)
+    );
+  }, [countryDataQuery, countryIso3DataQuery]);
+
+  useEffect(() => {
+    // expand all items only after the first load
+    if (!expandAll && !isLoading) setExpandAll(true);
+  }, [isLoading]);
+
+  const { countryDataList, countryIso3DataList } = useMemo(() => {
+    return CountryComparisonOperations.getFilteredCountryData(countryDataQuery, countryIso3DataQuery);
```

Comment:
This could be done in one liner without using {} and return but directly () => blabla